parser/transcript/src/main/java/de/danoeh/antennapod/parser/transcript/JsonTranscriptParser.java

```diff
@@ -33,13 +50,37 @@ public static Transcript parse(String jsonStr) {
                 }
                 duration += endTime - startTime;
 
+                prevSpeaker = speaker;
                 speaker = jsonObject.optString("speaker");
+                speakers.add(speaker);
+                if (StringUtils.isEmpty(speaker) && StringUtils.isNotEmpty(prevSpeaker)) {
+                    speaker = prevSpeaker;
+                }
                 String body = jsonObject.optString("body");
-                segmentBody += body + " ";
+                if (!prevSpeaker.equals(speaker)) {
+                    if (StringUtils.isNotEmpty(segmentBody)) {
+                        segmentBody = StringUtils.trim(segmentBody);
+                        transcript.addSegment(new TranscriptSegment(segmentStartTime,
+                                segmentEndTime,
+                                segmentBody,
+                                prevSpeaker));
+                        segmentStartTime = startTime;
+                        segmentBody = body.toString();
+                        duration = 0L;
+                        continue;
+                    }
+                }
 
-                if (duration >= TranscriptParser.MIN_SPAN) {
+                segmentBody += body;
+                if (duration >= TranscriptParser.MIN_SPAN
+                        && ((i + 1) <= objSegments.length()
+                        && StringUtils.isAlphanumeric(
```

Comment:
This alphanumeric check seems to prevent any merging of segments as soon as they use punctuation or similar

Reply:
I think what you should check is if the segment *starts* with an alphanumeric character. Not if consists entirely of alphanumeric characters. Also, there should probably be a limit. If a segment takes longer than, say, 8 seconds, it should not be merged -- even if that means it will start with a non-alphanumeric character. I was testing with this podcast: https://logbuch-netzpolitik.de/abonnieren

Reply:
fixed, thanks for the test case